tests/strategies/external/test_storage.py

```diff
@@ -93,6 +127,82 @@ def test_bid_aggregator_succeeds_with_warning_if_dof_are_disabled(external_stora
             "The following arguments are not supported for this market and have been removed from "
             "your order: ['requirements', 'attributes'].")
 
+    @staticmethod
+    def test_bid_aggregator_places_future_bid(external_storage, future_markets):
+        future_energy_kWh = 0.2
+        external_storage.area._markets.future_markets = future_markets
+
+        for time_slot in future_markets.market_time_slots:
+            external_storage.state.energy_to_buy_dict[time_slot] = future_energy_kWh
+            external_storage.state.offered_buy_kWh[time_slot] = 0.1
+            external_storage.state.pledged_buy_kWh[time_slot] = 0.1
+            return_value = external_storage.trigger_aggregator_commands(
+                {
+                    "type": "bid",
+                    "price": 200.0,
+                    "energy": future_energy_kWh,
+                    "time_slot": time_slot.format(DATE_TIME_FORMAT),
+                    "transaction_id": str(uuid.uuid4())
+                }
+            )
+            assert return_value["status"] == "ready"
+            bid_id = json.loads(return_value["bid"])["id"]
+            assert future_markets.bids[bid_id].energy == future_energy_kWh
+        assert len(future_markets.bids.values()) == len(future_markets.market_time_slots)
+
+    @staticmethod
+    def test_bid_aggregator_fails_placing_future_bid_more_than_required_energy(
```

Comment:
NIT (which applies to all method names in this PR). We have a convention for unit test method names [here](https://gridsingularity.atlassian.net/wiki/spaces/D3A/pages/2406449155/Naming+and+Style+conventions#Unittests) . Following this, I would expect another `_fails` here. Would you agree?

Reply:
Thanks. Will fix it.